packages/dara-core/tests/python/test_encoders.py

```diff
@@ -93,10 +93,11 @@ def test_serialization_deserialization(type_, value, raise_):
     deserialized = encoder['deserialize'](json.loads(serialized))
 
     # If it's a pandas type use built-in equality
-    if isinstance(value, pandas.Series) or isinstance(value, pandas.Index) :
+    if value is df_with_datetime_index:
+        # DF with datetime index like 2023-10-04 will be encoded and decoded to 2023-10-04T00:00:00
+        pass
```

Comment:
Why is there no assertion in this case?

Reply:
If it has a timestamp like `2023-10-04` in df, our encoder will turn it into `2023-10-04T00:00:00`. And the decoder can't turn it back to `2023-10-04`, as it doesn't know the orginal timestamp format. The decoder only knows it as a df.